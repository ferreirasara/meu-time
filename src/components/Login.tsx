import { Button, Card, Form, FormItemProps, Input, Typography, theme } from 'antd';
import { useState } from 'react';
import { BASE_URL } from '../utils/constants';
import { getHeaders } from '../utils/utils';
import { BarChartOutlined, LoginOutlined } from '@ant-design/icons';

export const Login = () => {
  const [loading, setLoading] = useState<boolean>(false);
  const [validateStatus, setValidateStatus] = useState<FormItemProps["validateStatus"]>("");
  const [validateMessage, setValidadeMessage] = useState<string>("");
  const {
    token: { colorBgContainerDisabled },
  } = theme.useToken();

  const onLogin = async (values: { apiKey: string }) => {
    if (!values?.apiKey) {
      setValidateStatus("error");
      setValidadeMessage("Por favor informe sua API Key");
      return;
    }

    setLoading(true);
    setValidateStatus("validating");
    setValidadeMessage("");

    try {
      const response = await fetch(BASE_URL + "/status", {
        headers: getHeaders(values.apiKey),
        method: "GET",
        redirect: 'follow'
      });

      const responseJson = await response?.json();

      if (!!responseJson?.response?.account) {
        setValidateStatus("success");
        localStorage.setItem("api-key", values.apiKey);
        window.location.reload();
      } else {
        setValidateStatus("error");
        setValidadeMessage("Sua API Key está incorreta. Verifique e tente novamente");
      }

    } catch (e) {
      console.log(e);
    }

    setLoading(false);
  };

  return <div
    className="login-container"
    style={{
      width: '100%',
      height: '100vh',
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      backgroundColor: colorBgContainerDisabled,
    }}
  >
    <Card
      title={
        <Typography.Title
          level={2}
          style={{ textAlign: 'center' }}
        >
          <BarChartOutlined /> Meu Time
        </Typography.Title>
      }
      bordered
    >
      <Form
        labelCol={{ span: 6 }}
        wrapperCol={{ span: 24 }}
        onFinish={onLogin}
      >
        <Form.Item
          label="API Key"
          name="apiKey"
          validateStatus={validateStatus}
          help={validateMessage}
        >
          <Input.Password />
        </Form.Item>

        <Form.Item>
          <Button
            type="primary"
            htmlType="submit"
            loading={loading}
            icon={<LoginOutlined />}
            block
          >
            Login
          </Button>
        </Form.Item>
      </Form>
      <div>
        <Typography.Text>
          Você pode obter uma key clicando <Typography.Link href='https://dashboard.api-football.com/register' target='_blank'>aqui</Typography.Link>.
        </Typography.Text>
      </div>
    </Card>
  </div>;
}