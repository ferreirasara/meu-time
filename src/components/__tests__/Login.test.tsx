import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { Login } from '../Login';

describe('Login', () => {
  beforeAll(() => {
    window.matchMedia = (query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // deprecated
      removeListener: jest.fn(), // deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })
  })

  it('match snapshot', () => {
    const login = renderer.create(<Login />).toJSON();
    expect(login).toMatchSnapshot();
  });
})