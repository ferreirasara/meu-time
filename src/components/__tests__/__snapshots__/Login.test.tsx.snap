// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login match snapshot 1`] = `
<div
  className="login-container"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(0, 0, 0, 0.04)",
      "display": "flex",
      "height": "100vh",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <div
    className="ant-card ant-card-bordered css-dev-only-do-not-override-15rg2km"
  >
    <div
      className="ant-card-head"
      style={Object {}}
    >
      <div
        className="ant-card-head-wrapper"
      >
        <div
          className="ant-card-head-title"
        >
          <h2
            className="ant-typography css-dev-only-do-not-override-15rg2km"
            style={
              Object {
                "WebkitLineClamp": undefined,
                "textAlign": "center",
              }
            }
          >
            <span
              aria-label="bar-chart"
              className="anticon anticon-bar-chart"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="bar-chart"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M888 792H200V168c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v688c0 4.4 3.6 8 8 8h752c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm-600-80h56c4.4 0 8-3.6 8-8V560c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v144c0 4.4 3.6 8 8 8zm152 0h56c4.4 0 8-3.6 8-8V384c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v320c0 4.4 3.6 8 8 8zm152 0h56c4.4 0 8-3.6 8-8V462c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v242c0 4.4 3.6 8 8 8zm152 0h56c4.4 0 8-3.6 8-8V304c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v400c0 4.4 3.6 8 8 8z"
                />
              </svg>
            </span>
             Meu Time
          </h2>
        </div>
      </div>
    </div>
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <form
        className="ant-form ant-form-horizontal css-dev-only-do-not-override-15rg2km"
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <div
          className="ant-form-item css-dev-only-do-not-override-15rg2km ant-form-item-with-help"
        >
          <div
            className="ant-row ant-form-item-row css-dev-only-do-not-override-15rg2km"
            style={Object {}}
          >
            <div
              className="ant-col ant-col-6 ant-form-item-label css-dev-only-do-not-override-15rg2km"
              style={Object {}}
            >
              <label
                className=""
                htmlFor="apiKey"
                title="API Key"
              >
                API Key
              </label>
            </div>
            <div
              className="ant-col ant-col-24 ant-form-item-control css-dev-only-do-not-override-15rg2km"
              style={Object {}}
            >
              <div
                className="ant-form-item-control-input"
              >
                <div
                  className="ant-form-item-control-input-content"
                >
                  <span
                    className="ant-input-affix-wrapper ant-input-password css-dev-only-do-not-override-15rg2km"
                    onClick={[Function]}
                  >
                    <input
                      className="ant-input css-dev-only-do-not-override-15rg2km"
                      disabled={false}
                      hidden={null}
                      id="apiKey"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      type="password"
                      value=""
                    />
                    <span
                      className="ant-input-suffix"
                    >
                      <span
                        aria-label="eye-invisible"
                        className="anticon anticon-eye-invisible ant-input-password-icon"
                        onClick={[Function]}
                        onMouseDown={[Function]}
                        onMouseUp={[Function]}
                        role="img"
                        tabIndex={-1}
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="eye-invisible"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                          />
                          <path
                            d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                          />
                        </svg>
                      </span>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ant-form-item css-dev-only-do-not-override-15rg2km"
        >
          <div
            className="ant-row ant-form-item-row css-dev-only-do-not-override-15rg2km"
            style={Object {}}
          >
            <div
              className="ant-col ant-col-24 ant-form-item-control css-dev-only-do-not-override-15rg2km"
              style={Object {}}
            >
              <div
                className="ant-form-item-control-input"
              >
                <div
                  className="ant-form-item-control-input-content"
                >
                  <button
                    className="ant-btn css-dev-only-do-not-override-15rg2km ant-btn-primary ant-btn-block"
                    disabled={false}
                    onClick={[Function]}
                    type="submit"
                  >
                    <span
                      className="ant-btn-icon"
                    >
                      <span
                        aria-label="login"
                        className="anticon anticon-login"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="login"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <defs>
                            <style />
                          </defs>
                          <path
                            d="M521.7 82c-152.5-.4-286.7 78.5-363.4 197.7-3.4 5.3.4 12.3 6.7 12.3h70.3c4.8 0 9.3-2.1 12.3-5.8 7-8.5 14.5-16.7 22.4-24.5 32.6-32.5 70.5-58.1 112.7-75.9 43.6-18.4 90-27.8 137.9-27.8 47.9 0 94.3 9.3 137.9 27.8 42.2 17.8 80.1 43.4 112.7 75.9 32.6 32.5 58.1 70.4 76 112.5C865.7 417.8 875 464.1 875 512c0 47.9-9.4 94.2-27.8 137.8-17.8 42.1-43.4 80-76 112.5s-70.5 58.1-112.7 75.9A352.8 352.8 0 01520.6 866c-47.9 0-94.3-9.4-137.9-27.8A353.84 353.84 0 01270 762.3c-7.9-7.9-15.3-16.1-22.4-24.5-3-3.7-7.6-5.8-12.3-5.8H165c-6.3 0-10.2 7-6.7 12.3C234.9 863.2 368.5 942 520.6 942c236.2 0 428-190.1 430.4-425.6C953.4 277.1 761.3 82.6 521.7 82zM395.02 624v-76h-314c-4.4 0-8-3.6-8-8v-56c0-4.4 3.6-8 8-8h314v-76c0-6.7 7.8-10.5 13-6.3l141.9 112a8 8 0 010 12.6l-141.9 112c-5.2 4.1-13 .4-13-6.3z"
                          />
                        </svg>
                      </span>
                    </span>
                    <span>
                      Login
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
      <div>
        <span
          className="ant-typography css-dev-only-do-not-override-15rg2km"
          style={
            Object {
              "WebkitLineClamp": undefined,
            }
          }
        >
          VocÃª pode obter uma key clicando 
          <a
            className="ant-typography css-dev-only-do-not-override-15rg2km"
            href="https://dashboard.api-football.com/register"
            rel="noopener noreferrer"
            style={
              Object {
                "WebkitLineClamp": undefined,
              }
            }
            target="_blank"
          >
            aqui
          </a>
          .
        </span>
      </div>
    </div>
  </div>
</div>
`;
